### PR TITLE
Few minor changes 

### PR DESCRIPTION
**Pull request by sbacchio**
Here are some small fixes needed for running quda within lyncs.

### CODE REVIEW

**Review by maddyscientist — APPROVED** [review]

**Comment by maddyscientist** [comment]
@mathiaswagner I think we can merge this once you have reviewed

File: QUDAConfig.cmake.in [hunk 1]
```diff
@@ -7,6 +7,10 @@ set(QUDA_MPI @QUDA_MPI@)
 set(QUDA_QIO @QUDA_QIO@)
 set(QUDA_OPENMP @QUDA_OPENMP@)
 set(QUDA_QDPJIT @QUDA_QDPJIT@)
+set(QUDA_VERSION @CMAKE_PROJECT_VERSION@)
+set(GITVERSION @GITVERSION@)
```
Comment (mathiaswagner) [inline QUDAConfig.cmake.in]: I guess we probably should rename this to be prefixed with `QUDA_` if we expose this.

Comment (sbacchio) [reply QUDAConfig.cmake.in]: Sure! FYI I need it for checking if a tunecache became obsolete

File: QUDAConfig.cmake.in [hunk 2]
```diff
@@ -7,6 +7,10 @@ set(QUDA_MPI @QUDA_MPI@)
 set(QUDA_QIO @QUDA_QIO@)
 set(QUDA_OPENMP @QUDA_OPENMP@)
 set(QUDA_QDPJIT @QUDA_QDPJIT@)
+set(QUDA_VERSION @CMAKE_PROJECT_VERSION@)
```
Comment (maddyscientist) [inline QUDAConfig.cmake.in]: This seems to introduce a new `QUDA_VERSION` variable which we already have defined in the source code.  What am I missing here?

Comment (sbacchio) [reply QUDAConfig.cmake.in]: Ok I see, I can get it from `quda_constants.h`. I missed that. Then it is not needed to be exposed here. I will remove it

File: QUDAConfig.cmake.in [hunk 3]
```diff
@@ -7,6 +7,9 @@ set(QUDA_MPI @QUDA_MPI@)
 set(QUDA_QIO @QUDA_QIO@)
 set(QUDA_OPENMP @QUDA_OPENMP@)
 set(QUDA_QDPJIT @QUDA_QDPJIT@)
+set(QUDA_GITVERSION @GITVERSION@)
```
Comment (mathiaswagner) [inline QUDAConfig.cmake.in:10]: Where is this being used then? Mostly curious.

File: CMakeLists.txt [hunk 4]
```diff
@@ -410,6 +410,7 @@ if(QUDA_DOWNLOAD_EIGEN)
     DOWNLOAD_ONLY
     YES)
   target_include_directories(Eigen SYSTEM INTERFACE ${Eigen_SOURCE_DIR})
+  install(DIRECTORY ${Eigen_SOURCE_DIR}/Eigen TYPE INCLUDE)
```
Comment (mathiaswagner) [inline CMakeLists.txt:413]: I am fine with this for now but technically Eigen is only used privately in QUDA and a calling application should not need it (the usual ones going through the interface should be fine). But maybe our headers are not set up for this yet. 
I guess if lyncs uses Eigen itself it should not use QUDA to get the headers.

**Review by mathiaswagner — APPROVED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (sbacchio) [reply CMakeLists.txt:413]: Ok I see what you mean. But actually, I had to add it due to an include error with the latest develop branch. Namely you have `eigen_helper.h` that is included by `eigensolve_quda.h` then included by `invert_quda.h`. So the installed headers are dependent on Eigen and without installing it I cannot load them.

Now, I have tried to remove the inclusion of `eigen_helper.h` and the first (and probably only) error I get is the following:
```
include/eigensolve_quda.h:201:61: error: unknown type name 'MatrixXi'
void permuteVecs(std::vector<ColorSpinorField> &kSpace, MatrixXi &mat, int size);
```
So as far as I can tell, currently, `Eigen` is not used only privately.

File: QUDAConfig.cmake.in (hunk 3 above)
Comment (sbacchio) [reply QUDAConfig.cmake.in:10]: The tuning uses the GITVERSION as a flag to declare when tunecache is obsolete. I then use it as well to create tuning directories that are dependent on it (unless the user specify its own `QUDA_RESOURCE_PATH`).

**Pull request merged**